src/commands.ts

```diff
@@ -1393,25 +1393,65 @@ export const addInlineComment = (
     const tr = state.tr.insert(pos, comment)
 
     if (from === to) {
-      const point = schema.nodes.highlight_marker.create({
-        id: comment.attrs.id,
-        tid: node.attrs.id,
-        position: 'point',
-      })
-      tr.insert(from, point)
-    } else {
-      const start = schema.nodes.highlight_marker.create({
-        id: comment.attrs.id,
-        tid: node.attrs.id,
-        position: 'start',
-      })
-      const end = schema.nodes.highlight_marker.create({
-        id: comment.attrs.id,
-        tid: node.attrs.id,
-        position: 'end',
-      })
-      tr.insert(from, start).insert(to + 1, end)
+      let start = from
+      let end = to
+      const resolvedPos = state.doc.resolve(from)
+      const blockStart = resolvedPos.start()
+      const blockEnd = resolvedPos.end()
+
+      // Move backward to find the start of the word
+      while (
+        start > blockStart &&
+        !/\s/.test(state.doc.textBetween(start - 1, start))
+      ) {
+        start--
+      }
+      // Move forward to find the end of the word
+      while (
+        end < blockEnd &&
+        !/\s/.test(state.doc.textBetween(end, end + 1))
+      ) {
+        end++
+      }
+
+      from = start
+      to = end
+
+      // If no word is found (cursor between spaces), search for the previous word
+      if (from === to) {
+        // Move backward through spaces
+        while (
+          start > blockStart &&
+          /\s/.test(state.doc.textBetween(start - 1, start))
+        ) {
+          start--
+        }
+        to = start
+
+        // Move backward to find the start of the previous word
+        while (
+          start > blockStart &&
+          !/\s/.test(state.doc.textBetween(start - 1, start))
+        ) {
+          start--
+        }
+
+        from = start
+      }
```

Comment:
Can we have this logic into its own function? something like `findWordBoundries`.